diff --git a/zebra/zebra_rib.c b/zebra/zebra_rib.c
index 0dc8a05e9..f10c94c1f 100644
--- a/zebra/zebra_rib.c
+++ b/zebra/zebra_rib.c
@@ -554,7 +554,8 @@ static int nexthop_active(afi_t afi, struct route_entry *re,
                        continue;
                }
 
-               if (match->type == ZEBRA_ROUTE_CONNECT) {
+               if (match->type == ZEBRA_ROUTE_CONNECT ||
+                   match->type == ZEBRA_ROUTE_KERNEL) {
                        /* Directly point connected route. */
                        newhop = match->ng.nexthop;
                        if (newhop) {
